src/full/Agda/TypeChecking/Irrelevance.hs

```diff
@@ -270,10 +261,12 @@ applyModalityToContextFunBody thing cont = do
 --   This is not the right thing to do when type checking interactively in a
 --   hole since it also marks all metas created during type checking as
 --   irrelevant (issue #2568).
+--
+--   Also set the current quantity to 0.
 wakeIrrelevantVars :: (MonadTCEnv tcm) => tcm a -> tcm a
 wakeIrrelevantVars
   = applyRelevanceToContextOnly Irrelevant
-  . applyQuantityToContextOnly  zeroQuantity
+  . applyQuantityToJudgementOnly zeroQuantity
 
```

Comment:
I would have just deleted the `. applyQuantityToContextOnly  zeroQuantity` line, was this change necessary for some tests?

Reply:
If this line is deleted, then you can't infer the type of an erased variable in the Emacs mode (in a non-erased context). This used to be possible. With this change you can also infer the type of an erased definition in a non-erased context.